mlir/test/Dialect/Vector/canonicalize/vector-transpose.mlir

```diff
@@ -247,3 +247,36 @@ func.func @negative_shape_cast_transpose(%arg : vector<6xi8>) -> vector<2x3xi8>
   %1 = vector.transpose %0, [1, 0] : vector<3x2xi8> to vector<2x3xi8>
   return %1 : vector<2x3xi8>
 }
+
+// -----
+
+// Test of transpose folding
```

Comment:
I have just noticed that you add this note to every test - thanks! Nit suggestion - why not add block comments, e.g.:
https://github.com/llvm/llvm-project/blob/0009a1783490a8ff69251a0ec7df1891a427cfb0/mlir/test/Dialect/Vector/vector-transfer-drop-unit-dims-patterns.mlir#L3-L5

The actual format is secondary - my main suggestion would be to avoid duplicating the same comment and to create "sections" to group similar tests together. Especially given that this file keeps growing 😅

Reply:
While working on https://github.com/llvm/llvm-project/pull/139706, I've noticed that these comments were incorrect/flipped (please double check). I prepared https://github.com/llvm/llvm-project/pull/139699 to address that. 

I am happy to re-factor the format (yours is more descriptive), or wait for you to update things instead. Please let me know if you have a preference :)

Reply:
Totally fine, I actually expected you to land it with your new format.. I should have communicated better yesterday.